src/stores/logFileStore.ts

```diff
@@ -0,0 +1,166 @@
+import useContextStore from "./contextStore";
+import useLogExportStore, {LOG_EXPORT_STORE_DEFAULT} from "./logExportStore";
+import useLogFileManagerProxyStore from "./logFileManagerProxyStore";
+import useQueryStore from "./queryStore";
+import useUiStore from "./uiStore";
+import useViewStore, {VIEW_STORE_DEFAULT} from "./viewStore";
+
+
+interface LogFileValues {
+    fileName: string;
+    fileSrc: Nullable<FileSrcType>;
+    numEvents: number;
+    onDiskFileSizeInBytes: number;
+}
+
+interface LogFileActions {
+    setFileName: (newFileName: string) => void;
+    setNumEvents: (newNumEvents: number) => void;
+    setOnDiskFileSizeInBytes: (newOnDiskFileSizeInBytes: number) => void;
+
+    loadFile: (fileSrc: FileSrcType, cursor: CursorType) => void;
+}
+
+type LogFileState = LogFileValues & LogFileActions;
+
+const LOG_FILE_STORE_DEFAULT: LogFileValues = {
+    fileName: "",
+    fileSrc: null,
+    numEvents: 0,
+    onDiskFileSizeInBytes: 0,
+};
+
+/**
+ * Format popup message shown when a structured log is loaded without a format string.
+ */
+const FORMAT_POP_UP_MESSAGE: PopUpMessage = Object.freeze({
+    level: LOG_LEVEL.INFO,
+    message: "Adding a format string can enhance the readability of your" +
+                    " structured logs by customizing how fields are displayed.",
+    timeoutMillis: LONG_AUTO_DISMISS_TIMEOUT_MILLIS,
+    title: "A format string has not been configured",
+});
+
+/**
+ * Handles export progress and append chunks to log file.
+ *
+ * @param logs
+ */
+const handleExportChunk = (logs: string) => {
+    const {logExportManager, setExportProgress} = useLogExportStore.getState();
+
+    if (null !== logExportManager) {
+        const progress = logExportManager.appendChunk(logs);
+        setExportProgress(progress);
+    }
+};
+
+/**
+ * Handles query results by updating the query progress and merging the results.
+ *
+ * @param progress
+ * @param results
+ */
+const handleQueryResults = (progress: number, results: QueryResults) => {
+    const {clearQueryResults, mergeQueryResults, setQueryProgress} = useQueryStore.getState();
+
+    setQueryProgress(progress);
+    if (QUERY_PROGRESS_VALUE_MIN === progress) {
+        clearQueryResults();
+
+        return;
+    }
+    mergeQueryResults(results);
+};
+
+const useLogFileStore = create<LogFileState>((set, get) => ({
+    ...LOG_FILE_STORE_DEFAULT,
+    loadFile: (fileSrc: FileSrcType, cursor: CursorType) => {
+        const {setFileName, setOnDiskFileSizeInBytes} = get();
+        const {postPopUp} = useContextStore.getState();
+        const {setExportProgress} = useLogExportStore.getState();
+        const {logFileManagerProxy} = useLogFileManagerProxyStore.getState();
+        const {clearQuery} = useQueryStore.getState();
+        const {setUiState} = useUiStore.getState();
+        const {isPrettified, setLogData, updatePageData} = useViewStore.getState();
+
+        setFileName(LOG_FILE_STORE_DEFAULT.fileName);
```

Comment:
This should be shown as "loading" when a file is being loaded, which is the existing behaviour.
```suggestion
        setFileName("Loading...");
```